src/praisonai/praisonai/_approval_spec.py

```
@@ -0,0 +1,134 @@
+
+
+def _parse_timeout(timeout_val: Optional[Union[str, int, float]]) -> Optional[float]:
+    """Parse timeout value to float, handling 'none' case."""
+    if timeout_val is None:
+        return None
+    if isinstance(timeout_val, str) and timeout_val.lower() == 'none':
+        return None
+    try:
+        return float(timeout_val)
+    except (ValueError, TypeError):
+        raise ValueError(f"Invalid timeout value: {timeout_val}")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)